src/main/webapp/3Dscript.js

```diff
@@ -178,6 +320,55 @@ function gui() {
   gui.add(poseTransform, 'scale', .5, 2, .25)
       .onChange(plotOrientation)
       .onFinishChange(plotTrajectory).name('Pose Scale Multiplier');
+
+  /* Adds index of point manipulation to maxgui. */
+  const max = poseLength;
+  oldEnd=poseLength;
+  gui.add(poseStartIndex, 'start', 0, max, 1)
+      .onFinishChange(hideOrientation);
+
+  gui.add(poseEndIndex, 'end', 0, max, 1)
+      .onFinishChange(hideOrientation);
+
+  /* Time value. */
+  time = gui.add(timeStart, 'time').onFinishChange(findTime);
+  /* Lat value. */
+  const latStart= {lat: ''};
+  lat = gui.add(latStart, 'lat');
+  /* lng value. */
+  const lngStart = {lng: ''};
+  lng = gui.add(lngStart, 'lng');
+  /* Alt value. */
+  const altStart = {alt: ''};
+  alt = gui.add(altStart, 'alt');
+  /* Yaw value. */
+  const yawStart= {yaw: ''};
+  yaw = gui.add(yawStart, 'yaw');
+  /* Pitch value. */
+  const pitchStart= {pitch: ''};
+  pitch = gui.add(pitchStart, 'pitch');
+  /* Roll value. */
+  const rollStart = {roll: ''};
+  roll = gui.add(rollStart, 'roll');
+}
+
+/** Updates time gui to show if time typed in is found or not. */
+function findTime() {
+  let found = false;
+  console.log(time.getValue());
+  for (let i = 0; i< poseLength; i++) {
+    // If point with time stamp is found
+    if (pose[i].gpsTimestamp == time.getValue()) {
+      unselectCylinder();
+      selectedIndex = i;
+      selectCylinder();
+      displayPointValues(i);
+      found= true;
+    }
+  }
```

Comment:
You can use binary search to improve the performance.
You can find the data which timestamp >= input timestamp instead of finding the exactly the same timestamp, since usually the user just want to search an ambiguous time and find the nearby data.

Reply:
ooo ok , i implemented binary search for exact time in the new commit, but will look into allowing ambiguous range now